app/controllers/devise/second_factor_webauthn_credentials_controller.rb

```diff
@@ -52,9 +62,21 @@ def verify_and_save_security_key(security_key_from_params)
       )
     end
 
+    # The default url to be used after creating a second factor key. You can overwrite
+    # this method in your own SecondFactorWebauthnCredentialsController.
+    def after_create_path
+      new_second_factor_webauthn_credential_path(resource_name)
+    end
+
     # The default url to be used after creating a second factor key. You can overwrite
     # this method in your own SecondFactorWebauthnCredentialsController.
     def after_update_path
+      request.referer || new_second_factor_webauthn_credential_path(resource_name)
+    end
+
+    # The default url to be used after deleting a second factor key. You can overwrite
+    # this method in your own SecondFactorWebauthnCredentialsController.
+    def after_destroy_path
```

Comment:
We should update our controller template with these changes too 👀 

https://github.com/cedarcode/devise-webauthn/blob/34682287deb3d865f4c5d93f419cf37460425464/lib/generators/devise/webauthn/templates/controllers/second_factor_webauthn_credentials_controller.rb.tt